include/zephyr/dt-bindings/pinctrl/ch32v20x_30x-pinctrl.h

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2024 MASSDRIVER EI (massdriver.space)
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __CH32V20X_V30X_PINCTRL_H__
+#define __CH32V20X_V30X_PINCTRL_H__
+
+#define CH32V20X_V30X_PINMUX_PORT_PA 0
+#define CH32V20X_V30X_PINMUX_PORT_PB 1
+#define CH32V20X_V30X_PINMUX_PORT_PC 2
+#define CH32V20X_V30X_PINMUX_PORT_PD 3
+#define CH32V20X_V30X_PINMUX_PORT_PE 4
+
+/*
+ * Defines the starting bit for the remap field.
+ */
+#define CH32V20X_V30X_PINMUX_SPI1_RM    0
+#define CH32V20X_V30X_PINMUX_I2C1_RM    1
+#define CH32V20X_V30X_PINMUX_USART1_RM  2
+#define CH32V20X_V30X_PINMUX_USART2_RM  3
+#define CH32V20X_V30X_PINMUX_USART3_RM  4
+#define CH32V20X_V30X_PINMUX_TIM1_RM    6
+#define CH32V20X_V30X_PINMUX_TIM2_RM    8
+#define CH32V20X_V30X_PINMUX_TIM3_RM    10
+#define CH32V20X_V30X_PINMUX_TIM4_RM    12
+#define CH32V20X_V30X_PINMUX_CAN1_RM    13
+#define CH32V20X_V30X_PINMUX_PD01_RM    15
+#define CH32V20X_V30X_PINMUX_TIM5CH4_RM 16
+#define CH32V20X_V30X_PINMUX_ETH_RM     21
+#define CH32V20X_V30X_PINMUX_CAN2_RM    22
+#define CH32V20X_V30X_PINMUX_RMII_RM    23
+#define CH32V20X_V30X_PINMUX_SDI_RM     24
+#define CH32V20X_V30X_PINMUX_SPI3_RM    28
+
+#define CH32V20X_V30X_PINMUX_TIM8_RM    (32 + 2)
+#define CH32V20X_V30X_PINMUX_TIM9_RM    (32 + 3)
+#define CH32V20X_V30X_PINMUX_TIM10_RM   (32 + 5)
+#define CH32V20X_V30X_PINMUX_USART4_RM  (32 + 16)
+#define CH32V20X_V30X_PINMUX_USART5_RM  (32 + 18)
+#define CH32V20X_V30X_PINMUX_USART6_RM  (32 + 20)
+#define CH32V20X_V30X_PINMUX_USART7_RM  (32 + 22)
+#define CH32V20X_V30X_PINMUX_USART8_RM  (32 + 24)
+#define CH32V20X_V30X_PINMUX_USART1_RM1 (32 + 26)
+
+/* Port number with 0-4 */
+#define CH32V20X_V30X_PINCTRL_PORT_SHIFT    0
+/* Pin number 0-15 */
+#define CH32V20X_V30X_PINCTRL_PIN_SHIFT     3
+/* Base remap bit 0-31 */
+#define CH32V20X_V30X_PINCTRL_RM_BASE_SHIFT 7
+/* Remap Register ID */
+#define CH32V20X_V30X_PINCTRL_PCFR_ID_SHIFT 12
+/* Function remapping ID 0-3 */
+#define CH32V20X_V30X_PINCTRL_RM_SHIFT      13
```

Comment:
If wanted, all mask-related operation can be done by only defining the mask, from which the LSB and MSB can be extracted via macros. For instance:
`#define CH32V20X_V30X_PINCTRL_PORT_MASK GENMASK(0, 2)`:

https://github.com/zephyrproject-rtos/zephyr/blob/2dcb61858b2cb2294bedca3b556fe210213a63d5/include/zephyr/sys/util.h#L75-L80

i.e. extracting a bit field, loading a value into a bit field...

Reply:
Done.